### ISSUE TITLE
clang warnings: unused command line arguments

### ISSUE THREAD

**Issue by Profpatsch** [tweag/rules_haskell#527]
In the Darwin build, we currently get these two warnings for each C library:

```
INFO: From HaskellHsc2hs tests/hsc/_hsc/hsc/BinHsc.hs:
clang-5.0: warning: argument unused during compilation: '-L/nix/store/lxi68avn1z2ihd4xxma4wl7y5ws19sg3-libc++-5.0.2/lib' [-Wunused-command-line-argument]
clang-5.0: warning: argument unused during compilation: '-L/nix/store/8g81q7hhnf4rh7l93xx17zx5v8x8v9b1-objc4-osx-10.11.6/lib' [-Wunused-command-line-argument]
clang-5.0: warning: argument unused during compilation: '-L/nix/store/lxi68avn1z2ihd4xxma4wl7y5ws19sg3-libc++-5.0.2/lib' [-Wunused-command-line-argument]
clang-5.0: warning: argument unused during compilation: '-L/nix/store/8g81q7hhnf4rh7l93xx17zx5v8x8v9b1-objc4-osx-10.11.6/lib' [-Wunused-command-line-argument]
```

Not sure whether that’s an upstream bug.

**Comment by aherrmann** [comment]
These warnings can be silenced by adding `-Wno-unused-command-line-argument` to [`cc-toolchain.nix`](https://github.com/tweag/rules_haskell/blob/82481879f522fd4643361f1fb53c8c8c605276c8/nixpkgs/cc-toolchain.nix#L23). Their origin is [the same file](https://github.com/tweag/rules_haskell/blob/82481879f522fd4643361f1fb53c8c8c605276c8/nixpkgs/cc-toolchain.nix#L28-L29). Was there a time when these `-L` flag did not cause these warnings on MacOS?

**Comment by Profpatsch** [comment]
> Was there a time when these `-L` flag did not cause these warnings on MacOS?

I don’t know.

These two lines introduce the warnings:
https://github.com/tweag/rules_haskell/blob/82481879f522fd4643361f1fb53c8c8c605276c8/nixpkgs/cc-toolchain.nix#L28-L29

I’m assuming a subset of targets might need them. Can we pass them only to that subset. Might not be worth the effort.

**Comment by tenevdev** [comment]
I think this could be closed as a result of https://github.com/tweag/rules_haskell/commit/dfc042294d8eb8a746bec1644dc9ab27b30b4a48 and https://github.com/tweag/rules_nixpkgs/blob/be7af4c9e5114f6262eef860c110bc7d3bd97009/toolchains/cc/cc.nix#L32

**Comment by aherrmann** [comment]
@tenevdev Good catch, agreed.
